### ISSUE TITLE
[css-color-adjust] [css-color-adjust] Initial value of the color property.

### ISSUE THREAD

**Issue by emilio** [w3c/csswg-drafts#9274]
Consider this test-case ([live](https://www.software.hixie.ch/utilities/js/live-dom-viewer/saved/11963)):

```html
<!doctype html>
<html style="color-scheme:dark">Root
  <div style="color:red">child div
    <span style="color: initial">what color is this span?</span>
    <span style="color: initial; color-scheme: light">And this?</span>
  </div>
</html>
```

Browsers disagree on what the behavior is:

 * In Firefox and Safari, the initial color value is still black.
 * In Chrome, the initial color behaves like an explicit `CanvasText`.

Right now the specs specify neither behavior.

From https://drafts.csswg.org/css-color-4/#the-color-property:

> Initial: CanvasText

Initial values are generally computed values, which means that you use the default color scheme, which comes from the `<meta name=color-scheme>` tag, absent in this test-case.

But then from https://drafts.csswg.org/css-color-adjust/#color-scheme-effect:

> On the root element, the [used color scheme](https://drafts.csswg.org/css-color-adjust/#used-color-scheme) additionally must affect the surface color of the [canvas](https://www.w3.org/TR/CSS21/intro.html#canvas), the initial value of the [color](https://drafts.csswg.org/css-color-4/#propdef-color) property, and the viewport’s scrollbars.

Which I understand it as "the second span should still have white text".

Test-case with `<meta name=color-scheme>` [here](https://www.software.hixie.ch/utilities/js/live-dom-viewer/saved/11964) for reference.

I think all browser's behavior make some sense in a way, and are reasonably easy to implement (Chromium's behavior _seems_ to be some sort of special-case where they treat `initial` much like `CanvasText`). But the spec behavior is rather weird.

I think Chromium's behavior is the most intuitive and would be ok resolving to do that, but happy to hear other opinions.

@lilles it seems this comes from a647cfd6a8 which seems directly from your spec draft. Is the spec behavior being different from Chromium's behavior intentional? I'm guessing not? :)

cc: @smfr @jfkthame @svgeesus @tabatkins

**Comment by tabatkins** [comment]
> Initial values are generally computed values, which means that you use the default color scheme, which comes from the <meta name=color-scheme> tag, absent in this test-case.

Wait, can you explain this more? I'm not quite sure I follow your reasoning.

> Which I understand it as "the second span should still have white text".

Note that the Color Adjust text is from 2019, predating [the commit that changed 'color''s initial value from black to CanvasText](https://github.com/w3c/csswg-drafts/commit/9da27b512b1c06ae1ea046135235a72f090002f4) which is from 2020. I think you're just getting confused by some dated text, meant to be an override at the time, which should now be removed since the initial value achieves what was intended more directly.

**Comment by emilio** [comment]
> Wait, can you explain this more? I'm not quite sure I follow your reasoning.

This is probably an implementation detail (though I recall WebKit working the same way in the past at least?). The way `initial` / `reset` are implemented are by pulling from a shared `ComputedStyle` object, since it's a computed style, has system colors already resolved.

I think this is the only case where `initial` needs a specified value. That said this is not unimplementable / hard to do, it's just "weird" compared to all other structs.

> I think you're just getting confused by some dated text, meant to be an override at the time, which should now be removed since the initial value achieves what was intended more directly.

Well the dated text is doing something different from `CanvasText`, that is my point.

**Comment by lilles** [comment]
One piece of history here might be that the system keywords were specified to compute to themselves at some point, and that the resolved/used value would take color-scheme into account. In that case it would be obvious to me that the root element and first span should have white color and the second span black color.

I think the "initial value of the color property" statement predates the CanvasText initial value.

This spec text from css-color-adjust also looks like it's assuming system keywords compute to themselves since it's talking about affecting the *used* value:

> For all elements, the user agent must match the following to the [used color scheme](https://drafts.csswg.org/css-? 
> color-adjust/#used-color-scheme):
> - the used value of the [system colors](https://drafts.csswg.org/css-color-4/#css-system-colors)

In short, I think the Blink behavior makes sense, but we probably need to do some adjustments in the specs.

**Comment by svgeesus** [comment]
> Note that the Color Adjust text is from 2019, predating [the commit that changed 'color''s initial value from black to CanvasText](https://github.com/w3c/csswg-drafts/commit/9da27b512b1c06ae1ea046135235a72f090002f4) which is from 2020. I think you're just getting confused by some dated text, meant to be an override at the time, which should now be removed since the initial value achieves what was intended more directly.

Done.

**Comment by svgeesus** [comment]
From CSS Color 4, [14.1. Resolving sRGB values](https://drafts.csswg.org/css-color-4/#resolving-sRGB-values)

> If the sRGB color was explicitly specified by the author as a [named color](https://drafts.csswg.org/css-color-4/#named-color), or as a [system color](https://drafts.csswg.org/css-color-4/#css-system-colors), the [specified value](https://drafts.csswg.org/css-cascade-5/#specified-value) is that named or system color, converted to [ASCII lowercase](https://infra.spec.whatwg.org/#ascii-lowercase). The computed and used value is the corresponding sRGB color, paired with the specified alpha channel (after clamping to [0.0, 1.0]) and defaulting to opaque if unspecified).

So, remove this?

> the used value of the [system colors](https://drafts.csswg.org/css-color-4/#css-system-colors)

**Comment by svgeesus** [comment]
@emilio all good now?

**Comment by svgeesus** [comment]
@emilio I believe this is all correct now, do you agree? If so the issue can be closed.

**Comment by emilio** [comment]
Yes
